hs/src/Reach/Eval.hs

```diff
@@ -159,31 +159,53 @@ didYouMean invalidStr validOptions maxClosest = case validOptions of
     closest = take maxClosest $ sortBy (comparing distance) validOptions
     distance = restrictedDamerauLevenshteinDistance defaultEditCosts invalidStr
 
-showVS :: Show a => a -> a -> String
-showVS fx fy = show fx <> " vs " <> show fy
-
-showDiff :: Eq b => a -> a -> (a -> b) -> String -> (b -> b -> String) -> String
-showDiff x y f lab s =
+showDiff :: Eq b => a -> a -> (a -> b) -> (b -> b -> String) -> String
+showDiff x y f s =
   let fx = f x
       fy = f y
    in case fx == fy of
         True -> ""
-        False -> "\n  " <> lab <> ": " <> s fx fy
+        False -> "\n  * " <> s fx fy
+
+getCorrectGrammer :: Foldable t => t a -> p -> p -> p
+getCorrectGrammer xs sing plur = case length xs of
+  1 -> sing
+  _ -> plur
 
 showStateDiff :: SLState -> SLState -> String
 showStateDiff x y =
-  showDiff x y st_mode "Mode" showVS
-    <> showDiff x y st_live "Live" showVS
-    <> showDiff x y st_after_first "After First Message" showVS
-    <> showDiff x y st_pdvs "Participant Definitions" showVS
+  "\nIssues:"
+  <> showDiff x y st_mode (\ xMode yMode ->
+    unwords ["Expected to be in ", show yMode, ", but in ", show xMode <> "."])
+  <> showDiff x y st_live (\ xLive yLive ->
+      case (xLive, yLive) of
+        (False, True) -> "Expected there to be live state."
+        (True, False) -> "Expected there to be no live state."
+        _             -> fail "Compiler error"
```

Comment:
```
λ> fail "Compiler error" :: String
""
```

This is probably not what you meant. Use `expect_throw` instead.

Reply:
`impossible`